content/doc/upgrade-guide/2.121.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: documentation
+title:  Jenkins Upgrade Guide
+notitle: true
+---
+
+== Upgrading to Jenkins LTS 2.121.x
+
+Each section covers the upgrade from the previous LTS release, the section on 2.121.1 covers the upgrade from 2.107.3.
+
+=== Upgrading to Jenkins LTS 2.121.1
+
+==== Remoting update
+
+https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md#320[Remoting changelog],
+https://issues.jenkins-ci.org/browse/JENKINS-50237[JENKINS-50237]
+
+The remoting library has been updated to improve diagnosability and reliability.
```

Comment:
The Remoting library got updated in order to ensure that exceptions are always properly serialized over the channel. If some of exception fields are blocked by JEP-200 class filters, there will be a sanitized exception delivered over the channel.

In order to get this fix applied, Remoting needs to be updated to version 3.13 or above on the agent side. Without the upgrade some remote operations on agents may fatally fail if the call returns exception, even if a plugin is ready to handle the exception and recover for it.

The following agent types should be updated:

* Java Web Start agents with `agent.jar` / `slave.jar` JARs on the disk. Version 3.19 or above is required
* Swarm Plugin Client. Version 3.11 or above is required
* Agent Docker Packages offered by the Jenkins project (jenkins/slave and jenkins/jnlp-slave). Version 3.19-1 or above is required

SSH Slaves based agents will always use the latest `agent.jar`, but other agent types may need to have their `agent.jar` updated manually.